tests/libstorage/storageclass.go

```diff
@@ -121,17 +118,30 @@ func CheckNoProvisionerStorageClassPVs(storageClassName string, numExpectedPVs i
 	// Verify we have at least `numExpectedPVs` available file system PVs
 	pvList, err := virtClient.CoreV1().PersistentVolumes().List(context.TODO(), metav1.ListOptions{})
 	Expect(err).ToNot(HaveOccurred())
+
+	if countLocalStoragePVAvailableForUse(pvList, storageClassName) < numExpectedPVs {
+		Skip("Not enough available filesystem local storage PVs available, expected: %d", numExpectedPVs)
+	}
+}
+
+func countLocalStoragePVAvailableForUse(pvList *k8sv1.PersistentVolumeList, storageClassName string) int {
 	count := 0
 	for _, pv := range pvList.Items {
-		if pv.Spec.StorageClassName != storageClassName || pv.Spec.NodeAffinity == nil || pv.Spec.NodeAffinity.Required == nil || len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) == 0 || (pv.Spec.VolumeMode != nil && *pv.Spec.VolumeMode == k8sv1.PersistentVolumeBlock) {
-			// Not a local volume filesystem PV
-			continue
-		}
-		if pv.Spec.ClaimRef == nil {
+		if isLocalPV(pv, storageClassName) && isPVAvailable(pv) {
 			count++
 		}
 	}
-	if count < numExpectedPVs {
-		Skip("Not enough available filesystem local storage PVs available, expected: %d", numExpectedPVs)
-	}
+	return count
+}
+
+func isLocalPV(pv k8sv1.PersistentVolume, storageClassName string) bool {
+	return pv.Spec.StorageClassName == storageClassName &&
```

Comment:
we can even extract storageclass name to the expression on line 130:
```
		if pv.Spec.StorageClassName == storageClassName && isLocalPV(pv) && isPVAvailable(pv) {
```
Sorry for not commenting on it before, but the more you clean the code, the more I can understand it and find pieces that do not fit ;).